src/main/controllers/JudgmentDetailsController.ts

```diff
@@ -0,0 +1,125 @@
+import { Response } from 'express';
+
+import { AppRequest } from '../definitions/appRequest';
+import { PageUrls, TranslationKeys } from '../definitions/constants';
+import { FormContent } from '../definitions/form';
+import { HubLinkStatus } from '../definitions/hub';
+import { AnyRecord } from '../definitions/util-types';
+import { getLogger } from '../logger';
+
+import { updateDecisionState, updateJudgmentNotificationState } from './helpers/CaseHelpers';
+import { createDownloadLink, getDocumentAdditionalInformation } from './helpers/DocumentHelpers';
+import { getPageContent } from './helpers/FormHelpers';
+import {
+  findSelectedDecision,
+  findSelectedJudgment,
+  getApplicationOfDecision,
+  getDecisionDetails,
+  getDecisions,
+  getJudgmentDetails,
+} from './helpers/JudgmentHelpers';
+import { getApplicationDocDownloadLink, getResponseDocDownloadLink } from './helpers/TseRespondentApplicationHelpers';
+
+const logger = getLogger('JudgmentDetailsController');
+export default class JudgmentDetailsController {
+  public get = async (req: AppRequest, res: Response): Promise<void> => {
+    const userCase = req.session.userCase;
+    req.session.documentDownloadPage = PageUrls.JUDGMENT_DETAILS;
+
+    const translations: AnyRecord = {
+      ...req.t(TranslationKeys.JUDGMENT_DETAILS, { returnObjects: true }),
+      ...req.t(TranslationKeys.COMMON, { returnObjects: true }),
+    };
+
+    let selectedDecision = undefined;
+    let decisions = undefined;
+    let header = undefined;
+    let selectedAttachment = undefined;
+    let selectedDecisionApplication = undefined;
+    let responseDocDownloadLink = undefined;
+    let selectedApplicationDocDownloadLink = undefined;
+
+    let selectedJudgment = undefined;
+    if (userCase?.sendNotificationCollection?.length) {
+      selectedJudgment = findSelectedJudgment(userCase.sendNotificationCollection, req.params.appId);
+    }
+
+    if (selectedJudgment === undefined) {
```

Comment:
A little complex of an if-else, see if you can break it apart a little.

Reply:
Moved some of this to helper